controllers/apps/transformer_component_workload.go

```diff
@@ -658,6 +666,50 @@ func (r *componentWorkloadOps) leaveMember4ScaleIn() error {
 	return err // TODO: use requeue-after
 }
 
+func unableToConnect(err error) bool {
+	if err == nil {
+		return false
+	}
+	if strings.Contains(err.Error(), "i/o timeout") {
+		return true
+	}
+	return false
+}
+
+// Try to leave `podToLeave` by pods in `desiredPods`,
+// if any error occurs not due to `unableToConnect` to pods in `desiredPods`, return it immediately.
+func (r *componentWorkloadOps) leaveMemberByOtherPods(desiredPods []*corev1.Pod, podToLeave *corev1.Pod) error {
+	parameters := make(map[string]any)
+	parameters["podName"] = podToLeave.Spec.Hostname
+
+	for _, pod := range desiredPods {
+		lorryCli, err1 := lorry.NewClient(*pod)
```

Comment:
It should follow the API definition, otherwise there will be risks in executing the action on other pods.